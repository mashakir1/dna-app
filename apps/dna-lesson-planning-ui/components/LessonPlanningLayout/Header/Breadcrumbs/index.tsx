import { Breadcrumb } from "./Breadcrumb";

import styles from "./breadcrumbs.module.scss";

interface BreadcrumbType {
  title: string;
  active: boolean;
  completed: boolean;
}

export const Breadcrumbs = ({
  breadcrumbs,
}: {
  breadcrumbs: Array<BreadcrumbType>;
}) => (
  <nav className={`${styles.breadcrumbs} flex`}>
    {breadcrumbs.map(({ title, active, completed }) => (
      <Breadcrumb
        key={title}
        title={title}
        active={active}
        completed={completed}
        // TODO:  handle click
        handleClick={() => {
          console.log("Clicked breadcrumb");
        }}
      />
    ))}
  </nav>
);
