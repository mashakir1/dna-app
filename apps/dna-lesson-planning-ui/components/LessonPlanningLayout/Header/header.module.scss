.header {
  width: 100vw;
  padding: $spacing-xxl;
  font-weight: 600;
  background-color: white;
}
