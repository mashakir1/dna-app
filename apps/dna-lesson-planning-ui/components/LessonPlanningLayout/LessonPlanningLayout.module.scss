.page {
  background-color: $colour_background_grey;
  min-height: 100vh;
}

.container {
  max-width: 864px;
  margin: 0 auto;
  margin-top: $spacing-xxl;
  padding-bottom: $spacing-xxl;
}
