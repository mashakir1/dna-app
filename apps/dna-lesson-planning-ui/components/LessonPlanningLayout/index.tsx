// Imports
import { ReactNode } from "react";

// Components
import { Header } from "./Header";

// Styles
import styles from "./LessonPlanningLayout.module.scss";

const Layout = ({ children }: { children: ReactNode }) => (
  <div className={styles.page}>
    <Header />
    <div className={`${styles.container} flex column gap-l`}>{children}</div>
  </div>
);
export default Layout;
