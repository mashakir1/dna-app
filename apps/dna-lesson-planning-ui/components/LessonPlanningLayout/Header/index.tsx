// Imports
import useTranslation from "next-translate/useTranslation";

// Components
import { OutlineLinkButton } from "components/Buttons";
import { Breadcrumbs } from "./Breadcrumbs";

// Styles
import styles from "./header.module.scss";

const breadcrumbs = [
  {
    title: "choose-unit",
    active: true,
    completed: false
  },
  {
    title: "journey",
    active: false,
    completed: false
  },
  {
    title: "plan-lesson",
    active: false,
    completed: false
  },
  {
    title: "review",
    active: false,
    completed: false
  }
];

export const Header = () => {
  const { t } = useTranslation("common");
  return (
    <div className={`${styles.header} flex justify-between items-center`}>
      <Breadcrumbs breadcrumbs={breadcrumbs} />
      <OutlineLinkButton text={t("cancel")} href="/" />
    </div>
  );
};
