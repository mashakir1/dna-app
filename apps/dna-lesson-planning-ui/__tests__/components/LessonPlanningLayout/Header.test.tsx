import { render } from "@testing-library/react";

import { Header } from "components/LessonPlanningLayout/Header";

// TODO: MOCK SESSION DATA TO TEST NAVIGATE OF BREADCRUMBS
describe("Header", () => {
  it("should render the base header layout", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
