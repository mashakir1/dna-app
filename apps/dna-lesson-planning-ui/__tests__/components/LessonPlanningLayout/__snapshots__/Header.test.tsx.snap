// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the base header layout 1`] = `
<div>
  <div
    class="header flex justify-between items-center"
  >
    <nav
      class="breadcrumbs flex"
    >
      <p
        aria-label="choose-unit"
        class="small breadcrumb active "
      >
        choose-unit
      </p>
      <p
        aria-label="journey"
        class="small breadcrumb  "
      >
        journey
      </p>
      <p
        aria-label="plan-lesson"
        class="small breadcrumb  "
      >
        plan-lesson
      </p>
      <p
        aria-label="review"
        class="small breadcrumb  "
      >
        review
      </p>
    </nav>
    <span>
      cancel
    </span>
  </div>
</div>
`;
